app/models/volunteer.rb

```diff
@@ -176,4 +183,13 @@ def to_s
   def record_acceptance_changed
     self["#{acceptance_change_to_be_saved[1]}_at".to_sym] = Time.zone.now if will_save_change_to_acceptance?
   end
+
+  def external_changed?
+    will_save_change_to_external?
+  end
+
+  def handle_user_with_external_change
+    user.destroy if external?
+    user.restore if internal?
```

Comment:
You can also do `user.destroy if external? else user.restore`, or use a multi-line `if` / `else` ;-)

But I really wouldn't write it with two `if` statements because the two conditions exclude each other, but I won't realize that until I actually look at the definitions of `external?` and `internal?`